File: sumpositive/sumpositive.py
import sys
sys.path.append('/Users/dsagman/Code/github/python-test-framework')
import codewars_test as test
from icecream import ic

def positive_sum(arr):
    return sum(filter((lambda x: x>0), arr))



@test.describe("positive_sum")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(positive_sum([1,2,3,4,5]),15)
        test.assert_equals(positive_sum([1,-2,3,4,5]),13)
        test.assert_equals(positive_sum([-1,2,3,4,-5]),9)
        
    @test.it("returns 0 when array is empty")
    def empty_case():
        test.assert_equals(positive_sum([]),0)      
        
    @test.it("returns 0 when all elements are negative")
    def negative_case():
        test.assert_equals(positive_sum([-1,-2,-3,-4,-5]),0)

File: sumpositive/sumpositive.hs
import Test.Hspec

positiveSum :: [Int] -> Int
positiveSum = sum . filter (>0)




-- spec = hspec $ do
--   describe "positiveSum" $ do
--     it "works for some examples" $ do
--       positiveSum [1, 2, 3, 4, 5] `shouldBe` 15
--       positiveSum [1, -2, 3, 4, 5] `shouldBe` 13
--       positiveSum [-1, 2, 3, 4, -5] `shouldBe` 9
    
--     it "returns 0 if array is empty" $ do
--       positiveSum [] `shouldBe` 0
    
--     it "returns 0 if all elements are negative" $ do
--       positiveSum [-1, -2, -3, -4, -5] `shouldBe` 0


